docs/external-signers/README.md

```diff
@@ -0,0 +1,75 @@
+and AWS kms.
+
+## Internal signer
+
+This is the default mode of operation. There are two submodes for the internsl signer:
+1. Plaintext key (Not recomended for production). This is mostly for testing and for systems
+where read access to the filesystem is heavyly restricted.
+2. Sealed Key. This is the recomended way for production systems and internal signer. To 
+unseal the key there are two mechanisms: AWS secrets manager and manual unsealing. AWS
+secrets manager is the recomended mechanism for self-healing deployments in AWS. 
+
+
+## External signers
+
+### Yubikey (PIV)
+In PIV mode we use the authentication slot (0x9a) for our signing operations.
+The keys in this mode can be either generated by the yuibkey itself or 
+imported.
+
+As written you only need the yubikey PIN if using self-generated keys. If using
+imported keys it is mandatory to have the public key of the key.
+
+The performance of signing using yuibkeys is dependent on the key type AND
+the key lenght. The recomendations is to use P256 or  P384 keys . Signatures
+take around 90ms, and  130ms respectively for each signing operation. Ed215519
+keys are also supported, but they may cause issues with old openidc clients
+so they are not recomended yet (if you need openidc compatibility).
+ 
+#### Generating keys
+There is no user facing code to generate the keys. The recomended way is to
+use `ykman` to let the yubikey generate its keys
+
+generate an Ed25519Key:
+> ykman piv keys generate -a ED25519 9a -
+
+Generate an P384 key:
+> ykman piv keys generate -a ECCP384 9a -
+
+#### Configuration
+
+The yubikey configuraiton is defined as a URL. Where the hostname is the yubikey serial number,
+the username is the public key (PKIX DER [RFC 5280](https://www.rfc-editor.org/rfc/rfc5280.html#section-4.1), but base64url encoded),
+and the password is the yubikey PIN. The type is "yubipiv"
+
+For example:
+```
+  external_signer_config:
+     type: "yubipiv"
+     #location: "yubipiv://MCowBQYDK2VwAyEABUlG-f3cM5LkFIox_M4qeNdBMYv1rD71Z0SnEXNP_bY=:123456@32720973"
+     location: "yubipiv://32720973"
+```
+
+The Hostname (serial number) is mandatory.
+
+### AWS kms
+
+The second external signer mechanism is AWS kms. When using this mode the AWS credentials need protection
+and monitorin should be placed in cloudtrail for the use of the KMS keys. Today AWS logs do not include a way
+to map what was signed at each operation so monitoring of the keys depends on tracking the actual key use
+from the keymasterd logs into the kms logs. 
+
+#### Configuraiton:
+The type is "AWS" and the location value is the full arn of the kms key. The AWS credentials
+needed as fetched by the AWS SDK so they can be part the of running profile, environment variables,
+or ".aws/credentials" values.
+
+```
+   external_signer_config:
+     type: "AWS"
+     location: "arn:aws:kms:us-west-2:807646279115:key/1aad97ad-32b5-484b-93af-155eb23a92d5"
```

Comment:
Hopefully this ARN is not a real resource?

Reply:
not anymore... oops.